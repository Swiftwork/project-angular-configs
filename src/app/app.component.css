:host {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;

  h1 {
    max-width: var(--d_sm);
    color: var(--c_color_alpha);
    text-align: center;
  }
}
