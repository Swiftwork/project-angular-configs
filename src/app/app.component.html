<h1>{{ 'welcome' | translate: { version: 7 } }}</h1>

<ng-container #debugRef></ng-container>
