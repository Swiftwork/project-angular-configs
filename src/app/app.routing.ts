import { Routes } from '@angular/router';
import { ViewDataResolver } from '@evry/ng-core';
import { DefaultComponent } from './core/default/default.component';

/* Router */
export const APP_ROUTES: Routes = [
];

/* Define our providers */
export const APP_ROUTING_PROVIDERS: any[] = [

];
