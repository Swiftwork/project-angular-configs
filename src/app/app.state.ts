import { Injectable } from '@angular/core';
import { ENVIRONMENT } from '../environments/environment';

export type InternalStateType = {
  [key: string]: any,
};

declare let siteSettingsData: any;

/**
 * AppStateService is used to get common global settings from EPI that affects the whole site.
 * It is dependant on the IAngularAppInitializationConfig from BackendModels.Application.
 * To access the current AppState one should subscribe to the public appState variable.
 * Observable<any>
 * @example
 *  this.appStateService.appState.subscribe((state) => {});
 */
@Injectable()
export class AppState {

  public _state: InternalStateType = {};

  constructor() {
    /* Preloaded site common settings */
    if (typeof siteSettingsData !== 'undefined') {
      this._state = siteSettingsData;
      /* Always remove except in HMR mode */
      if (!ENVIRONMENT.HMR) {
        siteSettingsData = undefined;
        document.body.removeChild(document.getElementById('data-site-settings'));
      }
    }
  }

  /**
   * Already return a clone of the current state.
   */
  public get state() {
    return this._state = this._clone(this._state);
  }
  /**
   * Never allow mutation
   */
  public set state(value) {
    throw new Error('do not mutate the `.state` directly');
  }

  public get(prop?: any) {
    /**
     * Use our state getter for the clone.
     */
    const state = this.state;
    return state.hasOwnProperty(prop) ? state[prop] : null;
  }

  public set(prop: string, value: any) {
    /**
     * Internally mutate our state.
     */
    return this._state[prop] = value;
  }

  private _clone(object: InternalStateType) {
    /**
     * Simple object clone.
     */
    return JSON.parse(JSON.stringify(object));
  }
}
