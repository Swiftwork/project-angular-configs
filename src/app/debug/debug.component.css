:host {
  position: fixed;
  display: block;
  top: 3vr;
  right: 0;
  padding: 1vr;
  z-index: var(--z_debug);
  background-color: var(--c_light_alpha);
  transform: translateX(100%);
  transition: transform var(--t_animation);

  &[aria-expanded='true'] {
    transform: none;
  }

  form,
  form > :last-child {
    margin-bottom: 0;
  }

  .title,
  .reload,
  .c-text-input,
  .c-checkbox-input,
  .c-select-input {
    color: white;
  }

  ::ng-deep .c-checkbox-input--label::before {
    color: white;
  }

  .title {
    .reload {
      float: right;
    }
  }

  .trigger {
    position: absolute;
    top: 1vr;
    left: 0;
    padding: 0 1vr;
    color: white;
    border-bottom: 1px solid rgba(255, 255, 255, 0.1);
    background-color: var(--c_light_alpha);
    transform: rotate(-90deg) translate(-100%, -100%);
    transform-origin: 0 0;
  }
}
