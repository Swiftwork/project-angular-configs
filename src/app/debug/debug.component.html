<a class="trigger no-print"
  role="button"
  tabindex="0"
  (click)="onSettingsUpdate('expanded', expanded = !expanded)"
  (keydown.enter)="onSettingsUpdate('expanded', expanded = !expanded)">
  <i class="icon-expand"></i> Debug
</a>

<h4 class="title no-print">Control panel
  <a class="icon-reload reload"
    (click)="refresh()"
    (keydown.enter)="refresh()"></a>
</h4>
<form class="no-print">

  <c-select-input name="lang"
    placeholder="Choose language"
    [options]="languages"
    [ngModel]="language"
    (ngModelChange)="onSettingsUpdate('language', $event)">Languages</c-select-input>

  <c-text-input name="api"
    [ngModel]="api"
    (ngModelChange)="onSettingsUpdate('api', $event)">API</c-text-input>

  <c-text-input name="scenario"
    [ngModel]="scenario"
    (ngModelChange)="onSettingsUpdate('scenario', $event)">Scenario</c-text-input>

  <c-select-input name="customer"
    placeholder="Choose mock customer"
    [options]="users"
    [ngModel]="user"
    (ngModelChange)="onSettingsUpdate('user', $event)">Users</c-select-input>

  <c-text-input name="delay"
    [ngModel]="delay"
    (ngModelChange)="onSettingsUpdate('delay', $event)">Delay</c-text-input>

  <c-checkbox-input name="live"
    [ngModel]="live"
    (ngModelChange)="onSettingsUpdate('live', $event)">Live data (requires reload)</c-checkbox-input>
</form>
