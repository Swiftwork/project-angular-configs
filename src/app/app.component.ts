import {
  Component,
  ComponentFactoryResolver,
  HostBinding,
  ViewChild,
  ViewContainerRef,
} from '@angular/core';
import { Router } from '@angular/router';
import { IRouteAlias, ModalComponent, RouterX } from '@evry/ng-core';
import { TranslateService } from '@ngx-translate/core';
import { Subscription } from 'rxjs';

import { ENVIRONMENT } from '../environments/environment';

/* SPECIAL CASE */
import { AppState } from './app.state';
import { DebugComponent } from './debug/debug.component';

declare let VERSION: string;
declare let routingAliasData: IRouteAlias;

@Component({
  selector: 'c-app',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css'],
})
export class AppComponent {
  @HostBinding('attr.app-version') version = VERSION;

  @ViewChild('debugRef', { read: ViewContainerRef }) debugRef: ViewContainerRef;

  @ViewChild('modalRef') modalRef: ModalComponent;

  private subs: Subscription[] = [];

  constructor(
    private router: Router,
    private translateService: TranslateService,
    private componentResolver: ComponentFactoryResolver,
    public appState: AppState,
  ) {
    this.translateService.setDefaultLang('sv');
    this.translateService.use('sv');

    if (
      ENVIRONMENT.DEBUG &&
      typeof sessionStorage.getItem('debug.language') === 'string'
    )
      this.translateService.use(sessionStorage.getItem('debug.language'));

    /* Preloaded routing alias */
    if (typeof routingAliasData !== 'undefined') {
      this.router.resetConfig(
        RouterX.AddRouteAlias(this.router.config, routingAliasData),
      );
      /* Always remove except in HMR mode */
      if (!ENVIRONMENT.HMR) {
        routingAliasData = undefined;
        document.body.removeChild(
          document.getElementById('data-routing-alias'),
        );
      }
    }
  }

  ngOnInit() {
    if (ENVIRONMENT.DEBUG) this.addDebugComponent();
  }

  addDebugComponent() {
    this.debugRef.createComponent(
      this.componentResolver.resolveComponentFactory(DebugComponent),
    );
  }

  ngOnDestroy() {
    this.subs.forEach(sub => sub.unsubscribe());
  }
}
