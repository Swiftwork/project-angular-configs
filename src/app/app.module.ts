import {
  Location,
  LocationStrategy,
  PathLocationStrategy,
  registerLocaleData,
} from '@angular/common';
import { HttpClient, HttpClientModule } from '@angular/common/http';
import localeSv from '@angular/common/locales/sv';
import { LOCALE_ID, NgModule } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { BrowserModule } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { RouterModule } from '@angular/router';
import { NgComponentsModule } from '@evry/ng-components';
import { NgCoreModule } from '@evry/ng-core';
import { TranslateLoader, TranslateModule } from '@ngx-translate/core';
import { TranslateHttpLoader } from '@ngx-translate/http-loader';

import { AppComponent } from './app.component';
import { APP_ROUTES, APP_ROUTING_PROVIDERS } from './app.routing';
import { AppState } from './app.state';
import { DebugModule } from './debug/debug.module';

/* LANGUAGES */
registerLocaleData(localeSv, 'sv');

declare let VERSION: string;

export function TranslateLoaderFactory(http: HttpClient) {
  return new TranslateHttpLoader(
    http,
    `${process.env.PUBLIC_PATH}i18n/`,
    `.json?version=${VERSION}`,
  );
}

@NgModule({
  imports: [
    /* @ANGULAR */
    BrowserModule,
    BrowserAnimationsModule,
    HttpClientModule,
    FormsModule,
    ReactiveFormsModule,
    TranslateModule.forRoot({
      loader: {
        provide: TranslateLoader,
        useFactory: TranslateLoaderFactory,
        deps: [HttpClient],
      },
    }),

    /* @EVRY */
    NgCoreModule.forRoot(),
    NgComponentsModule.forRoot(),

    /* Project */
    RouterModule.forRoot(APP_ROUTES),
    DebugModule,
  ],

  declarations: [AppComponent],

  /* Needed only for dynamic component creation such as dialogs and other containers */
  entryComponents: [
    // CONTAINER_DECLARATIONS,
    // INTERACTIVE_DECLARATIONS,
    // PRESENTATIONAL_DECLARATIONS,
    // COMPONENTS_DECLARATIONS,
  ],

  providers: [
    AppState,
    /* LOCALE */
    {
      provide: LOCALE_ID,
      useValue: 'sv-SE',
    },
    /* Location Strategy */
    Location,
    {
      provide: LocationStrategy,
      useClass: PathLocationStrategy,
    },
    APP_ROUTING_PROVIDERS,
  ],

  bootstrap: [AppComponent],
})
export class AppModule {}
