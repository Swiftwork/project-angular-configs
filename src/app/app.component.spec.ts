import { NO_ERRORS_SCHEMA } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { BrowserDynamicTestingModule } from '@angular/platform-browser-dynamic/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { TranslateModule } from '@ngx-translate/core';
import { AppComponent } from './app.component';
import { APP_ROUTES } from './app.routing';
import { AppState } from './app.state';
import { DebugComponent } from './debug/debug.component';

describe('AppComponent', () => {
  let fixture: ComponentFixture<AppComponent>;
  let component: AppComponent;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [
        TranslateModule.forRoot(),
        RouterTestingModule.withRoutes(APP_ROUTES),
      ],
      declarations: [AppComponent, DebugComponent],
      providers: [AppState],
      schemas: [NO_ERRORS_SCHEMA],
    });
    TestBed.overrideModule(BrowserDynamicTestingModule, {
      set: {
        entryComponents: [DebugComponent],
      },
    });
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(AppComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create the app', () => {
    expect(component).toBeTruthy();
  });

  it('should render markup that match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should dynamically add the debug controller', () => {
    const controller = fixture.debugElement.nativeElement.querySelector(
      'c-debug',
    );
    const trigger = controller.querySelector('.trigger');
    trigger.click();
    fixture.detectChanges();
    expect(controller.getAttribute('aria-expanded')).toBe('true');
  });
});
