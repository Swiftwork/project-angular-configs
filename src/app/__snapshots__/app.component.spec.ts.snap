// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should render markup that match snapshot 1`] = `
<c-app
  appState={[Function AppState]}
  componentResolver={[Function CodegenComponentFactoryResolver]}
  debugRef={[Function ViewContainerRef_]}
  modalRef="undefined"
  router={[Function Router]}
  subs={[Function Array]}
  translateService={[Function TranslateService]}
  version={[Function String]}
>
  <h1>
    welcome
  </h1><c-debug
    aria-expanded="false"
  >
    <a
      class="trigger no-print"
      role="button"
      tabindex="0"
    >
      <i
        class="icon-expand"
      />
       Debug
  
    </a>
    <h4
      class="title no-print"
    >
      Control panel 
      <a
        class="icon-reload reload"
      />
    </h4>
    <form
      class="no-print"
    >
      <c-select-input
        name="lang"
        placeholder="Choose language"
      >
        Languages
      </c-select-input>
      <c-text-input
        name="api"
      >
        API
      </c-text-input>
      <c-text-input
        name="scenario"
      >
        Scenario
      </c-text-input>
      <c-select-input
        name="customer"
        placeholder="Choose mock customer"
      >
        Users
      </c-select-input>
      <c-text-input
        name="delay"
      >
        Delay
      </c-text-input>
      <c-checkbox-input
        name="live"
      >
        Live data (requires reload)
      </c-checkbox-input>
    </form>
  </c-debug>
</c-app>
`;
